src/Bones/bone.ts

```diff
@@ -71,8 +70,9 @@ export class Bone extends Node {
 
     /** @hidden */
     set _matrix(value: Matrix) {
+        this._needToCompose = false; // in case there was a pending compose
         this._localMatrix.copyFrom(value);
-        this._needToDecompose = true;
+        this._markAsDirtyAndDecompose();

```

Comment:
I think we should check that `value.updateFlag` !== `this._localMatrix.updateFlag` to avoid unnecessary decomposition of the matrix when it is the same than the existing one.

The `Skeleton.prepare` method is doing:
```typescript
if (this._numBonesWithLinkedTransformNode > 0) {
    for (const bone of this.bones) {
        if (bone._linkedTransformNode) {
            // Computing the world matrix also computes the local matrix.
            bone._linkedTransformNode.computeWorldMatrix();
            bone._matrix = bone._linkedTransformNode._localMatrix;
            bone.markAsDirty();
        }
    }
}
```
So it is always updating the bone local matrix. If an animation is stopped, for eg, the matrices won't change so we would not need to re-set/re-decompose them. Note that I think we should also remove the `bone.markAsDirty()` call as it is now already done in the `_matrix` setter.

Reply:
> I think we should check that value.updateFlag !== this._localMatrix.updateFlag to avoid unnecessary decomposition of the matrix when it is the same than the existing one.

If value isn't the same object as this._localMatrix.updateFlag, I'm not sure this check is correct. Is there a way to check if the matrix are the same object?

Reply:
Wait, never mind, the flag is from a static, so it should be unique.

Reply:
Hmm, I tried this, but the check doesn't work for glTF for some reason. Investigating...

Reply:
I fixed it by copying the updateFlag when copying the matrix. This works, but I kind of think the copyFrom function should copy the updateFlag also.

Reply:
Waiting on #11162